src/main/flight/gps_rescue.c

```diff
@@ -362,20 +362,25 @@ static void rescueAttainPosition()
     const int zVelocityDerivative = zVelocityError - previousZVelocityError;
     previousZVelocityError = zVelocityError;
 
-    const int16_t hoverAdjustment = (hoverThrottle - 1000) / ct;
-    altitudeAdjustment = constrain(altitudeAdjustment + (throttle.Kp * zVelocityError + throttle.Ki * zVelocityIntegral + throttle.Kd * zVelocityDerivative),
-                                    gpsRescueConfig()->throttleMin - 1000 - hoverAdjustment, gpsRescueConfig()->throttleMax - 1000 - hoverAdjustment);
+    const int16_t hoverAdjustment = (int16_t)((hoverThrottle - 1000) / ct);
+
+    const float pterm = throttle.Kp * zVelocityError;
+    const float iterm = throttle.Ki * zVelocityIntegral;
+    const float dterm = throttle.Kd * zVelocityDerivative;
+    altitudeAdjustment = constrain(altitudeAdjustment + (int)(pterm + iterm  + dterm),
```

Comment:
To avoid truncation here, `altitudeAdjustment` needs to be changed from `int16_t` to `float`. This would make most sense if also changing `rescueThrottle` to `float`, however I don't know if this is necessary?

Reply:
Understand. Does it make sense?
```
altitudeAdjustment = constrain(altitudeAdjustment + lrintf(pterm + iterm  + dterm)
```

Reply:
constrainf ???

Reply:
```
static inline int constrain(int amt, int low, int high);
static inline float constrainf(float amt, float low, float high);
```
No. @mathiasvr is right here I think. It is `int` only.

Reply:
Yes but then it should use lrint ???

Reply:
No. `lrint` takes `double` which is bad.

Reply:
Yes correct. Saw that too. Thanks.

Reply:
`constrain()` will be good because throttle is in range of 0-1000

Reply:
Yeah as a note had to correct dshot as it was using floats here https://github.com/betaflight/betaflight/pull/10223